### PR TITLE
Improve integration tests for capturing the console log outputs

### LINKED ISSUE #1082: Underscores in resource methods escaped when there are special characters present
Consider the following path: `foo_bar_baz`

The resource method generated for this is:
```ballerina
resource function get foo_bar_baz() {}
```

But if we change the path slightly as follows: `foo_bar$_baz`

The generated method looks like
```ballerina
resource function get foo\_bar\$\_baz`() {}
```

As it can be seen, it's escaping the underscore unnecessarily. The characters to escape should conform to the following right? https://github.com/ballerina-platform/ballerina-lang/blob/17bc827c4defe81ebc24f3d8763905917052aabe/misc/identifier-util/src/main/java/io/ballerina/identifier/Utils.java#L38

### PR DESCRIPTION
**Pull request by lnash94**
## Purpose

- > Improve integration tests for IDL plugin
- Fix #1082 

## Automation tests
 - Unit tests  - Add
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: openapi-integration-tests/src/test/java/io/ballerina/openapi/TestUtil.java [hunk 1]
```diff
@@ -185,8 +182,8 @@ public static void deleteGeneratedFiles(String generatedFileName) throws IOExcep
     public static File[] getMatchingFiles(String project) throws IOException, InterruptedException {
         List<String> buildArgs = new LinkedList<>();
         //TODO: Change this function after fixing module name with client declaration alias.
-        boolean successful = executeRun(DISTRIBUTION_FILE_NAME, TEST_RESOURCE.resolve(project), buildArgs);
-        Assert.assertTrue(successful);
+        Process successful = executeRun(DISTRIBUTION_FILE_NAME, TEST_RESOURCE.resolve(project), buildArgs);
```
Comment (NipunaRanasinghe) [inline openapi-integration-tests/src/test/java/io/ballerina/openapi/TestUtil.java:185]:
```suggestion
        Process process = executeRun(DISTRIBUTION_FILE_NAME, TEST_RESOURCE.resolve(project), buildArgs);
```

File: openapi-integration-tests/src/test/java/io/ballerina/openapi/idl/client/IDLClientGenPluginNegativeTests.java [hunk 2]
```diff
@@ -42,41 +47,59 @@ public class IDLClientGenPluginNegativeTests extends OpenAPITest {
...
-
     @BeforeClass
     public void setupDistributions() throws IOException {
         TestUtil.cleanDistribution();
     }
 
     @Test(description = "Here uses the graphQl yaml as non openapi client")
     public void testNonOpenAPIClientDeclaration() throws IOException, InterruptedException {
-        boolean successful = executeRun(DISTRIBUTION_FILE_NAME, TEST_RESOURCE.resolve("project_07"),
+        Process successful = executeRun(DISTRIBUTION_FILE_NAME, TEST_RESOURCE.resolve("project_07"),
```
Comment (NipunaRanasinghe) [inline openapi-integration-tests/src/test/java/io/ballerina/openapi/idl/client/IDLClientGenPluginNegativeTests.java:57]:
```suggestion
        Process process = executeRun(DISTRIBUTION_FILE_NAME, TEST_RESOURCE.resolve("project_07"),
```

File: openapi-integration-tests/src/test/java/io/ballerina/openapi/idl/client/IDLClientGenPluginNegativeTests.java [hunk 3]
```diff
@@ -42,41 +47,59 @@ public class IDLClientGenPluginNegativeTests extends OpenAPITest {
...
-        boolean successful = executeRun(DISTRIBUTION_FILE_NAME, TEST_RESOURCE.resolve("project_12"),
+        Process successful = executeRun(DISTRIBUTION_FILE_NAME, TEST_RESOURCE.resolve("project_12"),
                 new ArrayList<>());
         File dir = new File(RESOURCE.resolve("client-idl-projects/project_12/generated/").toString());
         Assert.assertFalse(dir.exists());
+        String msg = " ERROR [main.bal:(2:5,2:34)] OpenAPI definition has errors:";
+        logComparison(successful, msg);
+    }
+
+    private static void logComparison(Process successful, String msg) throws IOException {
```
Comment (NipunaRanasinghe) [inline openapi-integration-tests/src/test/java/io/ballerina/openapi/idl/client/IDLClientGenPluginNegativeTests.java:95]:
```suggestion
    private static void logComparison(Process process, String msg) throws IOException {
```

Comment (NipunaRanasinghe) [inline openapi-integration-tests/src/test/java/io/ballerina/openapi/idl/client/IDLClientGenPluginNegativeTests.java:95]: `logComparison` is not a valid method name by the convention. Please use a proper name 
https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html#:~:text=Methods%20should%20be%20verbs%2C%20in,of%20each%20internal%20word%20capitalized.&text=Except%20for%20variables%2C%20all%20instance,words%20start%20with%20capital%20letters.

File: openapi-integration-tests/src/test/java/io/ballerina/openapi/idl/client/IDLClientGenPluginNegativeTests.java [hunk 4]
```diff
@@ -42,41 +47,59 @@ public class IDLClientGenPluginNegativeTests extends OpenAPITest {
...
+        try (BufferedReader br = new BufferedReader(new InputStreamReader(successful.getErrorStream()))) {
+            Stream<String> logLines = br.lines();
+            String generatedLog = logLines.collect(Collectors.joining("\n"));
+            logLines.close();
+            generatedLog = (generatedLog.trim()).replaceAll(WHITESPACE_PATTERN, "");
+            msg = (msg.trim()).replaceAll(WHITESPACE_PATTERN, "");
```
Comment (NipunaRanasinghe) [inline openapi-integration-tests/src/test/java/io/ballerina/openapi/idl/client/IDLClientGenPluginNegativeTests.java:101]:
```suggestion
            msg = msg.trim().replaceAll(WHITESPACE_PATTERN, "");
```

File: openapi-integration-tests/src/test/java/io/ballerina/openapi/idl/client/IDLClientGenPluginNegativeTests.java [hunk 5]
```diff
@@ -42,41 +47,59 @@ public class IDLClientGenPluginNegativeTests extends OpenAPITest {
...
+            Assert.assertTrue(generatedLog.contains(msg), "IDL client negative test fails.");
```
Comment (NipunaRanasinghe) [inline openapi-integration-tests/src/test/java/io/ballerina/openapi/idl/client/IDLClientGenPluginNegativeTests.java:102]:
```suggestion
            Assert.assertTrue(generatedLog.contains(msg), String.format("compiler output doesn't contain the expected output: %s",msg));
```

**Review by NipunaRanasinghe — CHANGES_REQUESTED** [review]

File: openapi-integration-tests/src/test/java/io/ballerina/openapi/idl/client/IDLClientGenPluginNegativeTests.java [hunk 6]
```diff
@@ -42,41 +47,60 @@ public class IDLClientGenPluginNegativeTests extends OpenAPITest {
...
     }
 
     @Test
     public void testInvalidSwaggerContract() throws IOException, InterruptedException {
-        boolean successful = executeRun(DISTRIBUTION_FILE_NAME, TEST_RESOURCE.resolve("project_12"),
+        Process process = executeRun(DISTRIBUTION_FILE_NAME, TEST_RESOURCE.resolve("project_12"),
                 new ArrayList<>());
         File dir = new File(RESOURCE.resolve("client-idl-projects/project_12/generated/").toString());
         Assert.assertFalse(dir.exists());
+        String msg = " ERROR [main.bal:(2:5,2:34)] OpenAPI definition has errors:";
```
Comment (NipunaRanasinghe) [inline openapi-integration-tests/src/test/java/io/ballerina/openapi/idl/client/IDLClientGenPluginNegativeTests.java:91]:
```suggestion
        String msg = "ERROR [main.bal:(2:5,2:34)] OpenAPI definition has errors:";
```

File: openapi-integration-tests/src/test/java/io/ballerina/openapi/idl/client/IDLClientGenPluginNegativeTests.java [hunk 7]
```diff
@@ -42,41 +47,60 @@ public class IDLClientGenPluginNegativeTests extends OpenAPITest {
...
+        compareLogOutput(process, msg);
+    }
+
+    private static void compareLogOutput(Process process, String msg) throws IOException {
```
Comment (NipunaRanasinghe) [inline openapi-integration-tests/src/test/java/io/ballerina/openapi/idl/client/IDLClientGenPluginNegativeTests.java:95]: Shall we have a name like below as we are only checking the error stream?
```suggestion
    private static void assertOnErrorStream(Process process, String msg) throws IOException {
```

Comment (lnash94) [reply openapi-integration-tests/src/test/java/io/ballerina/openapi/idl/client/IDLClientGenPluginNegativeTests.java:95]: +1

**Review by NipunaRanasinghe — APPROVED** [review]

**Pull request merged**